web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -473,12 +520,15 @@ private void subscribeToTerminationNotification() {
                             subscribeToTerminationNotification();
                         } else {
                             info.notifyConnectionError(Js.cast(fail));
+                            connectionLost();
                         }
                         return;
                     }
                     assert success != null;
 
                     // welp; the server is gone -- let everyone know
+                    connectionLost();
+
                     info.notifyConnectionError(new ResponseStreamWrapper.Status() {
```

Comment:
Removing the notifyConnectionError here, and instead going with a disconnect event exposes a missing case that we need to consider for DHC, how to signal that the server has shut down _and told the client this_, but how to potentially handle the server coming back again afterwards. We probably need a new event, or metadata on the disconnect event, but likewise should probably welcome the server back by prompting another login (right now we'll get the "reauth failed" event) by the user.

Reply:
Subscribers of TerminationNotification get a tombstone from the server with `abnormalTermination` set to `false`. We can probably use this to indicate a normal / expected shutdown? Though, I personally get these on my machines as the DH instances restart whenever they are redeployed. Either way, maybe this is the path you want to use?

Reply:
We do get "expected shutdown", but if you are testing an application or some server side debugging, you're going to edit, restart, edit, restart - an expected shutdown isn't the same as "never come back without user interaction".

Success or failure we can tell the client that the disconnect happened, but the question then is do we say "the server is down (here's the error|this was on purpose)" or just say "its down, it might be up again soon, please dismiss the message"?

Reply:
Added a new event on IdeConnection, EVENT_SHUTDOWN that will emit just the status message from the shutdown, rather than the full grpc status. It will only fire if the client gets notification that the server is explicitly going away, as opposed to losing that stream (or any other stream).